ocaml/xapi/xapi_ha_vm_failover.ml

```diff
@@ -16,6 +16,34 @@ module D = Debug.Make (struct let name = "xapi_ha_vm_failover" end)
 
 open D
 
+module VMGroupMap = Map.Make (struct
+  type t = [`VM_group] Ref.t
+
+  let compare = Ref.compare
+end)
+
+module IntMap = Map.Make (struct
+  type t = int
+
+  let compare = compare
+end)
+
+module HostsSet = Set.Make (struct
+  type t = int * int64 * [`host] Ref.t
+
+  let compare (vm_cnt_0, size_0, host_ref_0) (vm_cnt_1, size_1, host_ref_1) =
```

Comment:
Could get rid of the nested `match`:
```
match (Int.compare x1 x2, Int.compare y1 y2, Ref.compare z1 z2) with
| n, _, _ when n <> 0 -> n
| _, n, _ when n <> 0 -> n
| _, _, n -> n
```

Reply:
I think in this case, the nested `match` is very clear, please refer the [ocaml doc](https://ocaml.org/manual/4.13/api/Set.html#top) in this style.